integration-tests/src/test/java/com/datastax/oss/driver/core/metadata/TabletMapSchemaChangesIT.java

```diff
@@ -0,0 +1,178 @@
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.metadata.KeyspaceTableNamePair;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.metadata.schema.KeyspaceMetadata;
+import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata;
+import com.datastax.oss.driver.api.testinfra.CassandraSkip;
+import com.datastax.oss.driver.api.testinfra.ScyllaRequirement;
+import com.datastax.oss.driver.api.testinfra.ccm.CustomCcmRule;
+import com.datastax.oss.driver.api.testinfra.session.SessionRule;
+import com.datastax.oss.driver.api.testinfra.session.SessionUtils;
+import com.datastax.oss.driver.internal.core.loadbalancing.BasicLoadBalancingPolicy;
+import com.datastax.oss.driver.internal.core.metadata.schema.TabletMapSchemaChangeListener;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+@ScyllaRequirement(
+    minOSS = "6.0.0",
+    minEnterprise = "2024.2",
+    description = "Needs to support tablets")
+@CassandraSkip(description = "Tablets are ScyllaDB-only extension")
+// Ensures that TabletMap used by MetadataManager behaves as desired on certain events
+public class TabletMapSchemaChangesIT {
+
+  // Same listener as the one registered on initialization by
+  // DefaultDriverContext#buildSchemaChangeListener
+  // for TabletMap updates. Note that this mock only verifies that it reacts to ".onXhappening()"
+  // calls and the
+  // actual working listener updates the TabletMap.
+  private static final TabletMapSchemaChangeListener listener =
+      Mockito.mock(TabletMapSchemaChangeListener.class);
+  private static final CustomCcmRule CCM_RULE =
+      CustomCcmRule.builder()
+          .withNodes(2)
+          .withCassandraConfiguration(
+              "experimental_features", "['consistent-topology-changes','tablets']")
+          .build();
+  private static final SessionRule<CqlSession> SESSION_RULE =
+      SessionRule.builder(CCM_RULE)
+          .withConfigLoader(
+              SessionUtils.configLoaderBuilder()
+                  .withDuration(DefaultDriverOption.REQUEST_TIMEOUT, Duration.ofSeconds(15))
+                  .withClass(
+                      DefaultDriverOption.LOAD_BALANCING_POLICY_CLASS,
+                      BasicLoadBalancingPolicy.class)
+                  .withBoolean(DefaultDriverOption.METADATA_SCHEMA_ENABLED, true)
+                  .withDuration(DefaultDriverOption.METADATA_SCHEMA_WINDOW, Duration.ofSeconds(0))
+                  .build())
+          .withSchemaChangeListener(listener)
+          .build();
+
+  @ClassRule
+  public static final TestRule CHAIN = RuleChain.outerRule(CCM_RULE).around(SESSION_RULE);
+
+  private static final int INITIAL_TABLETS = 32;
```

Comment:
@Bouncheck I think that in this test you don't need to set the initial tablets size, which will be better to use the defaults from the server side.